WalletWasabi.Gui/Controls/TransactionDetails/ViewModels/TransactionDetailsViewModel.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Linq;
+
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.TransactionBuilding;
+using WalletWasabi.Gui.Controls.TransactionDetails.Models;
+using WalletWasabi.Stores;
+
+namespace WalletWasabi.Gui.Controls.TransactionDetails.ViewModels
+{
+	public class TransactionDetailsViewModel : ReactiveObject
+	{
+		private int _confirmations;
+		private bool _confirmed;
+		private DateTimeOffset _dateTime;
+		private string _amountBtc;
+		private string _label;
+		private int _blockHeight;
+		private string _transactionId;
+		private string _totalInputValue;
+		private string _totalOutputValue;
+		private int _inputCount;
+		private int _outputCount;
+
+		public DateTimeOffset DateTime
+		{
+			get => _dateTime;
+			set => this.RaiseAndSetIfChanged(ref _dateTime, value);
+		}
+
+		public int Confirmations
+		{
+			get => _confirmations;
+			set => this.RaiseAndSetIfChanged(ref _confirmations, value);
+		}
+
+		public bool Confirmed
+		{
+			get => _confirmed;
+			set => this.RaiseAndSetIfChanged(ref _confirmed, value);
+		}
+
+		public string AmountBtc
+		{
+			get => _amountBtc;
+			set => this.RaiseAndSetIfChanged(ref _amountBtc, value);
+		}
+
+		public string Label
+		{
+			get => _label;
+			set => this.RaiseAndSetIfChanged(ref _label, value);
+		}
+
+		public int BlockHeight
+		{
+			get => _blockHeight;
+			set => this.RaiseAndSetIfChanged(ref _blockHeight, value);
+		}
+
+		public string TransactionId
+		{
+			get => _transactionId;
+			set => this.RaiseAndSetIfChanged(ref _transactionId, value);
+		}
+
+		public string TotalInputValue
+		{
+			get => _totalInputValue;
+			set => this.RaiseAndSetIfChanged(ref _totalInputValue, value);
+		}
+
+		public string TotalOutputValue
+		{
+			get => _totalOutputValue;
+			set => this.RaiseAndSetIfChanged(ref _totalOutputValue, value);
+		}
+
+		public int InputCount
+		{
+			get => _inputCount;
+			set => this.RaiseAndSetIfChanged(ref _inputCount, value);
+		}
+
+		public int OutputCount
+		{
+			get => _outputCount;
+			set => this.RaiseAndSetIfChanged(ref _outputCount, value);
+		}
+
+		public static TransactionDetailsViewModel FromBuildTxnResult(BitcoinStore stores, BuildTransactionResult result)
```

Comment:
We should be able to create this class from NBitcoin.PSTB. In TransactionBroadcaster we have that and because we are planning to use the TransactionDetails there too. 

So we should change the logic here to be able to support that too. 

https://github.com/molnard/WalletWasabi/blob/866a790019307a0343ef7d5f5271c1f6d019ce43/WalletWasabi.Gui/Controls/WalletExplorer/TransactionBroadcasterViewModel.cs#L187

Reply:
Fixed